app/src/test/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java

```diff
@@ -0,0 +1,263 @@
+    @Override
+    public void createPost(User user, Forum forum, Post post) throws ForumNotFoundException,
+        PostAlreadyExistingException {
+        if (!forumExists(forum.getGroup(), forum)) {
+            throw new ForumNotFoundException();
+        }
+        for (Group g : groups) {
+            if (g.getName().equals(forum.getGroup().getName())) {
+                for (Forum f : g.getForums()) {
+                    if (f.getName().equals(forum.getName())) {
+                        for (Post p : f.getPosts()) {
+                            if (p.getUsername().equals(post.getUsername())
+                                && p.getCreationDate().equals(post.getCreationDate())) {
+                                throw new PostAlreadyExistingException();
+                            }
+                        }
+                        f.addPost(post);
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void deletePost(User user, Forum forum, DateTime postCreationDate) throws ForumNotFoundException,
+        PostNotFoundException {
+        if (!forumExists(forum.getGroup(), forum)) {
+            throw new ForumNotFoundException();
+        }
+        for (Group g : groups) {
+            if (g.getName().equals(forum.getGroup().getName())) {
+                for (Forum f : g.getForums()) {
+                    if (f.getName().equals(forum.getName())) {
+                        boolean found = false;
+                        for (Post p : f.getPosts()) {
+                            if (p.getUsername().equals(user.getUsername())
+                                && p.getCreationDate().compareTo(postCreationDate) == 0) {
+                                found = true;
+                                f.removePost(user, postCreationDate);
+                            }
+                        }
+                        if (!found) {
+                            throw new PostNotFoundException();
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void updatePost(User user, Post post, String newText) throws ForumNotFoundException, PostNotFoundException {
+        Forum postForum = post.getForum();
+        Group postGroup = postForum.getGroup();
+        if (!forumExists(postGroup, postForum)) {
+            throw new ForumNotFoundException();
+        }
+        for (Group g : groups) {
+            if (g.getName().equals(postGroup.getName())) {
+                updateForums(user, post, newText, postForum, g);
+            }
+        }
+    }
+
+    /**
+     * Update the forums.
+     * @param user The user
+     * @param post The post
+     * @param newText The new text
+     * @param postForum The forum of the post
+     * @param group The group
+     * @throws PostNotFoundException The post is not found
+     */
+    private void updateForums(User user, Post post, String newText, Forum postForum, Group group)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method length is 17 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5421176)